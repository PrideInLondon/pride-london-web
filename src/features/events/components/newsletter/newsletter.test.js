import React from 'react'
import { shallow, mount } from 'enzyme'
import NewsletterForm from './'

describe('<NewsletterForm/>', () => {
  beforeAll(() => {
    global.___loader = {
      enqueue: jest.fn(),
    }
    global.fetch = jest.fn().mockImplementation(() => Promise.resolve({}))
  })
  afterAll(() => {
    global.___loader.enqueue.mockReset()
    global.fetch.mockClear()
    delete global.fetch
    wrapper.unmount()
  })

  it('matches snapshot', () => {
    const wrapper = shallow(<NewsletterForm />)
    expect(wrapper).toMatchSnapshot()
  })

  const wrapper = mount(<NewsletterForm />)
  const form = wrapper.find('form')
  const emailInput = wrapper.find('input[type="email"]')
  const submitButton = wrapper.find('newsletter__SubmitButton')

  it('should capture email enterred into the input field in the form action attibute', () => {
    emailInput.simulate('change', {
      target: { value: 'FOOBAR@email.com' },
    })
    expect(form.html()).toContain('value="FOOBAR@email.com"')
  })

  it('should fire a form submission when the Submit Button is clicked', () => {
    submitButton.simulate('submit')
    expect(global.fetch).toHaveBeenCalledTimes(1)
    expect(
      global.fetch
    ).toHaveBeenLastCalledWith(
      'https://prideinlondon.us6.list-manage.com/subscribe?u=8289d9ca2253b74574f849c73&id=a2423c3382&MERGE0=FOOBAR@email.com',
      { method: 'POST' }
    )
  })
})
