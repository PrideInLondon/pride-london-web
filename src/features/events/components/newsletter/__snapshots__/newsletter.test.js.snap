// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsletterForm/> matches snapshot 1`] = `
<Fragment>
  <newsletter__Heading>
    Don't miss our all year round events
  </newsletter__Heading>
  <newsletter__Form
    onSubmit={[Function]}
  >
    <newsletter__StyledInput
      handleChange={[Function]}
      id="email"
      label="E-mail address"
      required={true}
      type="email"
      value=""
    />
    <newsletter__SubmitButton
      fullmobile={true}
      primary={true}
      type="submit"
    >
      Join now
    </newsletter__SubmitButton>
  </newsletter__Form>
</Fragment>
`;
